slave.mk

```diff
@@ -581,8 +581,8 @@ $(addprefix $(DEBS_PATH)/, $(SONIC_ONLINE_DEBS)) : $(DEBS_PATH)/% : .platform \
 	# Load the target deb from DPKG cache
 	$(call LOAD_CACHE,$*,$@)
 
-	# Skip building the target if it is already loaded from cache
-	if [ -z '$($*_CACHE_LOADED)' ] ; then
+	# Skip building the target if it is already loaded from cache or exists in target/ directory
+	if [ -z '$($*_CACHE_LOADED)' ] && [ ! -e $(DEBS_PATH)/$* ] ; then
```

Comment:
It the target has already existed, it is expected that the target will not be built again.
One of the reason may be the modified time is older than the depended targets, for instance, .platform.

Another reason, it may be caused by dpkg cache side effect, when the dpkg cache enabled, it will add a new dependency file "%.dep", the file does not exist, so need to build the targe again.

Did you download the file in the target folder manually? Have you tried to touch the file? And create a .dep file?
It may be a workaround for you.
1. touch target/.../<xxx.dep>.dep
2. download target/.../<xxx.dep>, and touch target/.../<xxx.dep>

You change is LGTM, but one side effect is not to download the file again if the other depended files changed, based on the make system it should be done again.

Reply:
The problem is a difference in behavior when using dpkg cache versus not using it.

With standard build, the debs do _not_ get redownloaded if they already exist in the workspace, regardless of any timestamps. But with dpkg cache enabled, the debs seem to _always_ get redownloaded, regardless of the timestamps of any files (including `.dep` files).

This change just makes the behavior consistent so debs do not get redownloaded if they already exist under `target/`. This is in-line with the behavior of GNU Make as well--Make uses existence of output files to determine whether they need to be built or not. With this change, SONiC build behavior aligns with the standard expectation for Makefile-based build.

Reply:
I did a local test, it worked fine. I can see the timestamp not changed and not .log file generated, I built several times for the sample target as below:
```
-rw-r--r-- 1 xumia gxumia 45221328 Apr 19 03:13 target/debs/bullseye/libsaibcm_8.4.0.2_amd64.deb
```

Looks the dependency is good, only depended on .platform, and .dep file. I have no idea why it built again.
.platform \
	$(call dpkg_depend,$(DEBS_PATH)/%.dep)

But I am good for your change, approved, only curious why it gets rebuilt.

Reply:
Yeah, I'm not sure either... But the proposed change does work for me. Thank you!